frappe/email/doctype/email_queue/email_queue.py

```diff
@@ -30,6 +31,17 @@ def get_duplicate(self, recipients):
 		duplicate.set_recipients(recipients)
 		return duplicate
 
+	def update_db(self, commit=False, **kwargs):
+		frappe.db.set_value(self.DOCTYPE, self.name, kwargs)
+		if commit:
+			frappe.db.commit()
```

Comment:
Original PR doesn't have this commit 🤔 Why is this required?

IIRC we have had horrible bugs because of random commits in email.

Reply:
example: https://github.com/frappe/frappe/pull/13612

Reply:
> Original PR doesn't have this commit 🤔 Why is this required?

because [nightly](https://github.com/frappe/frappe/blob/2081f485c87cc9280b26344711b451bf82d85f99/frappe/email/doctype/email_queue/email_queue.py#L70) handles it

Reply:
still doesn't answer _why_ this commit is required?

Reply:
We commit at end of request/background job: 
https://frappeframework.com/docs/user/en/api/database#database-transaction-model